### PR TITLE
[stdlib] Fixing compilation warnings

### PR DESCRIPTION
**Pull request by moiseev**
<rdar://problem/30320630>

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please test

File: stdlib/public/SDK/Intents/INRequestRideIntent.swift [hunk 1]
```diff
@@ -25,7 +25,7 @@ extension INRequestRideIntent {
     paymentMethod: INPaymentMethod? = nil,
     scheduledPickupTime: INDateComponentsRange? = nil
   ) {
-    if #available(iOS 10.3, watchOS 3.2, *) {
+    if #available(iOS 10.3, *) {
```
Comment (parkera) [inline stdlib/public/SDK/Intents/INRequestRideIntent.swift]: Is it correct to drop availability here?

**Review by parkera — COMMENTED** [review]
Can we also apply the Foundation changes to the same files in swift-corelibs-foundation and make sure it compiles and works there?

File: stdlib/public/SDK/Intents/INRequestRideIntent.swift (hunk 1 above)
Comment (moiseev) [inline stdlib/public/SDK/Intents/INRequestRideIntent.swift]: The whole extension is only available in watchOS 3.2, so this check will always be true. (At least that what the compiler warning said).

**Comment by moiseev** [comment]
@parkera I looked at a couple of files in xcorelibs-foundation and they don't have this code. Will compile it locally and fix the warnings. Thanks for pointing it out!

**Comment by parkera** [comment]
FYI we are stuck with some warnings because the compiler produces warnings about bridging on Darwin in code that is required for Linux.

**Pull request merged**